lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -5852,7 +5852,18 @@ static ParseResult parseFPrintfAttrs(OpAsmParser &p,
 
 static void printFPrintfAttrs(OpAsmPrinter &p, Operation *op,
                               DictionaryAttr attr) {
-  printElideEmptyName(p, op, attr, {"formatString", "outputFile"});
+  printElideEmptyName(p, op, attr,
+                      {"formatString", "outputFile", "operandSegmentSizes"});
+}
+
+LogicalResult FPrintFOp::verify() {
+  for (auto operand : getOutputFileSubstitutions())
+    if (!operand.getDefiningOp<firrtl::HierarchicalModuleNameOp>())
+      return mlir::emitError(operand.getLoc())
+             << "Only a hierarchical module name can be used as a substitution "
+                "for fprintf output file name";
+
+  return success();
```

Comment:
Nice verifier. It didn't occur to me that the verifiers are much easier now given the fstring ops as this no longer requires re-parsing.